packages/medusa/src/strategies/batch-jobs/product/export.ts

```diff
@@ -0,0 +1,393 @@
+      {
+        limit,
+        offset,
+        order,
+        fields,
+        expand,
+      },
+      {
+        isList: true,
+        defaultRelations: this.defaultRelations_,
+      }
+    )
+
+    batchJob.context = {
+      ...(context ?? {}),
+      list_config: listConfig,
+      filterable_fields,
+    }
+
+    return batchJob
+  }
+
+  async processJob(batchJobId: string): Promise<void> {
+    let offset = 0
+    let limit = this.BATCH_SIZE
+    let advancementCount = 0
+    let productCount = 0
+
+    return await this.atomicPhase_(
+      async (transactionManager) => {
+        let batchJob = await this.batchJobService_
+          .withTransaction(transactionManager)
+          .retrieve(batchJobId)
+
+        const { writeStream, fileKey, promise } = await this.fileService_
+          .withTransaction(transactionManager)
+          .getUploadStreamDescriptor({
+            name: "product-export",
+            ext: "csv",
+          })
+
+        const header = await this.buildHeader()
+        writeStream.write(header)
+
+        offset = batchJob.context?.list_config?.skip ?? offset
+        limit = batchJob.context?.list_config?.take ?? this.BATCH_SIZE
+        advancementCount =
+          batchJob.result?.advancement_count ?? advancementCount
+
+        const { list_config = {}, filterable_fields = {} } = batchJob.context
+        const [productList, count] = await this.productService_
+          .withTransaction(transactionManager)
+          .listAndCount(filterable_fields, {
+            ...list_config,
+            skip: offset,
+            take: limit,
+          } as FindProductConfig)
+
+        productCount = count
+        let products: Product[] = productList
+
+        while (offset < productCount) {
+          if (!products?.length) {
+            products = await this.productService_
+              .withTransaction(transactionManager)
+              .list(filterable_fields, {
+                ...list_config,
+                skip: offset,
+                take: limit,
+              } as FindProductConfig)
+          }
+
+          products.forEach((product: Product) => {
+            const lines = this.buildProductVariantLines(product)
+            lines.forEach((line) => writeStream.write(line))
+          })
+
+          advancementCount += products.length
+          offset += products.length
+          products.length = 0
+
+          batchJob = await this.batchJobService_
+            .withTransaction(transactionManager)
+            .update(batchJobId, {
+              context: {
+                ...batchJob.context,
+                file_key: fileKey,
+                list_config: {
+                  ...list_config,
+                  skip: offset,
+                },
+              },
+              result: {
+                count: productCount,
+                advancement_count: advancementCount,
+                progress: advancementCount / productCount,
+                err: undefined,
```

Comment:
nit: currently the result shape has a field for `error` and `batchJobService.setFailed` sets `error` just want to make sure that you are aware of this.

Reply:
Oups, well spitted it is indeed a legacy from previous iteration. 👍